File: typescript/packages/bench/src/rules/statusSequenceRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class StatusSequenceRule implements RuleContract {
    readonly key = "status-sequence";

    evaluate(data: Dataset): boolean {
        return data.completedOrders().length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/status_sequence_rule.rs
//! Rule 18/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct StatusSequenceRule;

impl RuleContract for StatusSequenceRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.completed_orders().is_empty()
    }
}
